### PR TITLE
Handle nested outputs when calculating accuracy

### PR DESCRIPTION
**Pull request by kevinwuTT**
### Ticket
Issue was discovered because of inconsistent accuracy values between the pytest run and generated accuracy script https://github.com/tenstorrent/pytorch2.0_ttnn/issues/835

### Problem description
Some outputs are nested Dict types. For example, CLIP has an output that has the following keys:

```
CLIPOutput = {
    "logits_per_image",
    "logits_per_text",
    "text_embeds",
    "image_embeds",
    "text_model_output" : {
        "last_hidden_state",
        "pooler_output"
    },
    "vision_model_output" : {
        "last_hidden_state",
        "pooler_output"
    }
}
```
Currently, `calculate_accuracy` silently skips the nested outputs so only the first 4 items are factored into the calculation. 

### What's changed

- Recursively walk through each item including nested Dicts and Lists until torch tensors are found and calculate the mean of the accumulated pcc values.
- Include a unit test to demonstrate the bug
- Compare the whole output of dino model instead just one attribute

### CODE REVIEW

**Review by jmalone-tt — APPROVED** [review]
Looks good to me

**Review by ayerofieiev-tt — APPROVED** [review]
